### PR TITLE
API Documentation: Issue 367

### PR DESCRIPTION
**Pull request by DavidBrien**
### Definition of done:
- [x] The feature works -> production build
- [x] Tests exist and run successfully
- [x] All Tests work
- [x] Happy Path covered
- [x] No new Bugs/Followups
- [x] Implemented changes for the Groovy-API

- [x] Documentation:
  - Readme.md
  - Doc-Section in PIT
  - Groovy/OpenAPI

---

## Release Info

- Fix Documentation issues
- Add missing parameters
- Consistent code

### Release Types:

Groovy-API: Minor

### Release notes:
---
## Code Review Checklist:
- [x] Does the code work?
- [x] Is all the code easily understood?
- [x] Are the function signatures named properly (function name, parameter names?)
- [x] Is the spelling correct?

- [x]  Is there redundant or duplicate code?
- [x]  Is there any commented out code?
- [x]  Can any logging or debugging code be removed?
- [x]  Is `for(i)` or `while` used? If so, remove
- [x]  Can any of the code be replaced with library functions?

-  Do you check input parameters (such as route parameters) for:
- [x] Type
- [x] Length
- [x] Format
- [x] Range?

- [x] Can comments be removed?
- [x] Is "external functionality" (API, Readme.md etc) documented/updated?
- [x] Is any unusual behavior or edge case handling described?

- [x] Do tests exist?
- [x] Do the snapshots look correct?
- [x] Are tests comprehensive?
- [x] Do unit tests actually test that the code is performing the intended functionality?

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (UweStolz) [inline yarn.lock]: Remove the yarn.lock

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 2]
```diff
@@ -274,14 +253,27 @@ class PitGroovyApi {
      * @throws PIMUnreachableException
      * @throws PIMInternalErrorException
      */
-    public Response getClassificationGroupSubgroup(String classificationId, String classificationGroupId) {
+    public Response getClassificationGroupSubgroups(String classificationId, String classificationGroupId) {
```
Comment (UweStolz) [inline customizationArea/server/serviceMethods/pimApi.groovy]: `getClassificationGroupSubgroups` is plural, but we do only get a single subgroup

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 3]
```diff
@@ -274,14 +259,36 @@ class PitGroovyApi {
...
+     * @param  classificationId ClassificationId
+     * @param  options Optional
+     * <p>
+     * Key: limit - The field which should be used for a limit of how much items should returned. By default the value is 100.
+     * <p>
+     * <p>
+     * Key: offset - The field which should be used to determine how many items to skip at the beginning.
+     * <p>
+     * <p>
+     * Key: sort - The field which should be used for sorting. Available values are productId and statusId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: available fields are not productId and statusId..

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 4]
```diff
@@ -368,14 +398,58 @@ class PitGroovyApi {
...
         String path = productsByClassificationGroupPath(classificationId, classificationGroupId)
-        restGet(path, options)
+        restGet(path, query)
     }
 
     /**
      * Retrieve all Classifications
      * @param  options Optional
      * <p>
+     * Key: sort - The field which should be used for sorting. Available values are productId and statusId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: Available values are not productId and statusID

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 5]
```diff
@@ -627,12 +791,18 @@ class PitGroovyApi {
         def query = [:]
         def sort = options.sort
         def order = options.order
+        def limit = options.limit
+        def offset
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: offset will be null every time

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 6]
```diff
@@ -734,14 +957,61 @@ class PitGroovyApi {
      * @throws PIMInternalErrorException
      */
     public Response getAllProductsBySupplierId(String supplierId, options=[:]) {
+        def limit = options.limit
+        def offset = options.offset
+        def sort = options.sort
+        def order = options.order
+        def catalogId = options.catalogId
+        def filterBySupplierId = options.filterBySupplierId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: Remove, supplierId already in scope

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 7]
```diff
@@ -691,16 +867,63 @@ class PitGroovyApi {
      * @throws PIMUnreachableException
      * @throws PIMInternalErrorException
      */
-    public Response getAllProductsByCatalogId(String catalogId, options=[:]) {
+    public Response getAllProductsByCatalogId(String catalogId, options = [:]) {
+        def limit = options.limit
+        def offset = options.offset
+        def sort = options.sort
+        def order = options.order
+        def filterByCatalogId = options.filterByCatalogId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: remove, catalogId already in scope

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 8]
```diff
@@ -691,16 +867,63 @@ class PitGroovyApi {
...
+        def filterBySupplierId = options.filterBySupplierId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: rename to supplierId

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 9]
```diff
@@ -368,14 +398,58 @@ class PitGroovyApi {
...
      */
     public Response getAllProductsByClassificationGroup(String classificationId, String classificationGroupId, options=[:]) {
+        def limit = options.limit
+        def offset = options.offset
+        def sort = options.sort
+        def order = options.order
+        def catalogId = options.catalogId
+        def supplierId = options.supplierId
+        def contractId = options.contractId
+        def filterByClassificationId = options.filterByClassificationId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: remove

Comment (alexejFroebel) [reply customizationArea/server/serviceMethods/pimApi.groovy]: also remove classificationId from optionParams in description

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 10]
```diff
@@ -368,14 +398,58 @@ class PitGroovyApi {
...
+        if(catalogId){
+            query.put("catalogId", catalogId)
+        }
+        if(supplierId){
+            query.put("supplierId", supplierId)
+        }
+        if(contractId){
+            query.put("contractId", contractId)
+        }
+        if(filterByClassificationId){
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: remove

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 11]
```diff
@@ -583,9 +734,15 @@ class PitGroovyApi {
      * @throws PIMUnreachableException
      * @throws PIMInternalErrorException
      */
-    public Response getProductDocuments(String catalogId, String productId) {
+    public Response getProductDocuments(String catalogId, String productId, options = [:]) {
+        def languageIds = options.languageIds
+        def variantId = options.variantId
+
+        def query = [:]
+        query.put('languageIds', languageIds.join(','))
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: make an if around this, like everywhere else

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 12]
```diff
@@ -583,9 +734,15 @@ class PitGroovyApi {
...
+        query.put('variantId', variantId)
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: same

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 13]
```diff
@@ -691,16 +867,63 @@ class PitGroovyApi {
...
-    public Response getAllProductsByCatalogId(String catalogId, options=[:]) {
+    public Response getAllProductsByCatalogId(String catalogId, options = [:]) {
+        def limit = options.limit
+        def offset = options.offset
+        def sort = options.sort
+        def order = options.order
+        def filterByCatalogId = options.filterByCatalogId
+        def filterBySupplierId = options.filterBySupplierId
+        def contractId = options.contractId
+        def classificationId = options.classificationId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: param classificationGroupId is missing

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 14]
```diff
@@ -775,14 +1045,61 @@ class PitGroovyApi {
...
      */
     public Response getAllProductsByClassificationId(String classificationId, options=[:]) {
+        def limit = options.limit
+        def offset = options.offset
+        def sort = options.sort
+        def order = options.order
+        def catalogId = options.catalogId
+        def supplierId = options.supplierId
+        def contractId = options.contractId
+        def filterByClassificationId = options.filterByClassificationId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: remove, classificationId already in scope

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 15]
```diff
@@ -775,14 +1045,61 @@ class PitGroovyApi {
      * @throws PIMInternalErrorException
      */
     public Response getAllProductsByClassificationId(String classificationId, options=[:]) {
+        def limit = options.limit
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: missing param: classificationGroupId

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 16]
```diff
@@ -816,14 +1133,61 @@ class PitGroovyApi {
      * @throws PIMInternalErrorException
      */
     public Response getAllProductsByContractId(String contractId, options=[:]) {
+        def limit = options.limit
+        def offset = options.offset
+        def sort = options.sort
+        def order = options.order
+        def catalogId = options.catalogId
+        def supplierId = options.supplierId
+        def filterByContractId = options.filterByContractId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: remove

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 17]
```diff
@@ -816,14 +1133,61 @@ class PitGroovyApi {
      * @throws PIMInternalErrorException
      */
     public Response getAllProductsByContractId(String contractId, options=[:]) {
+        def limit = options.limit
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: classificationGroupId param missing

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 18]
```diff
@@ -380,7 +380,7 @@ class PitGroovyApi {
      * Key: contractId - If set, products will be filtered by this contractId
      * <p>
      * <p>
-     * Key: classificationId - If set, products will be filtered by this classificationId
+     * Key: classificationGroupId - If set, products will be filtered by this classificationGroupId
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: remove

File: customizationArea/server/serviceMethods/pimApi.groovy [hunk 19]
```diff
@@ -540,13 +533,6 @@ class PitGroovyApi {
 
     /**
      * Retrieve all attributeValues of a Product
-     * @param  options Optional
```
Comment (alexejFroebel) [inline customizationArea/server/serviceMethods/pimApi.groovy]: Why remove this?
